src/libraries/Common/tests/Tests/System/StringTests.cs

```diff
@@ -5312,12 +5312,40 @@ private static void ToLower_Culture(string input, string expected, CultureInfo c
             }
         }
 
+        public static IEnumerable<object[]> ToLower_Invariant_TestData()
+        {
+            yield return new object[] { "", "" };
+            yield return new object[] { "Ab", "ab" };
+            yield return new object[] { "H-/", "h-/" };
+            yield return new object[] { "Hello", "hello" };
+            yield return new object[] { "hElLo", "hello" };
+            yield return new object[] { "AbcdAbc", "abcdabc" };
+            yield return new object[] { "AbcdAbcd", "abcdabcd" };
+            yield return new object[] { "AbcdAbcd/", "abcdabcd/" };
+            yield return new object[] { "AbcdAbcd/-", "abcdabcd/-" };
+            yield return new object[] { "AbcdAbcd/-_", "abcdabcd/-_" };
+            yield return new object[] { "AbcdAbcd-bcdAbc", "abcdabcd-bcdabc" };
+            yield return new object[] { "AbcdAbcd-bcdAbcd", "abcdabcd-bcdabcd" };
+            yield return new object[] { "AbcdAbcd-bcdAbcdA", "abcdabcd-bcdabcda" };
+            yield return new object[] { "AbcdAbcd-bcdAbcdA/", "abcdabcd-bcdabcda/" };
+            // Non-ASCII char:
+            yield return new object[] { "\u436", "\u436" };
+            yield return new object[] { "H\u436/", "h\u436/" };
+            yield return new object[] { "Hell\u436", "hell\u436" };
+            yield return new object[] { "hEl\u436o", "hel\u436o" };
+            yield return new object[] { "AbcdAb\u436", "abcdab\u436" };
+            yield return new object[] { "Abcd\u436bcd", "abcd\u436bcd" };
+            yield return new object[] { "AbcdAbc\u436/", "abcdabc\u436/" };
+            yield return new object[] { "AbcdAbcd/\u436", "abcdabcd/\u436" };
+            yield return new object[] { "AbcdAbcd/-\u436", "abcdabcd/-\u436" };
+            yield return new object[] { "AbcdAbc\u436d-bcdAbc", "abcdabc\u436d-bcdabc" };
+            yield return new object[] { "AbcdAbcd-b\u436dAbcd", "abcdabcd-b\u436dabcd" };
+            yield return new object[] { "AbcdAbcd-bcd\u436bcdA", "abcdabcd-bcd\u436bcda" };
+            yield return new object[] { "AbcdAbcd-bcdAbc\u436A/", "abcdabcd-bcdabc\u436a/" };
```

Comment:
nit: add another case which has the lower case variant for non-ASCII chars: https://github.com/dotnet/runtime/blob/52eed7dc3912594631dfecca56da3057bec52ba6/src/libraries/System.Collections/tests/Generic/Dictionary/Dictionary.Tests.cs#L344

```suggestion
            yield return new object[] { "AbcdAbcd-bcdAbc\u0436A/", "abcdabcd-bcdabc\u0436a/" };
            yield return new object[]
            {
                "\u0430\u0431\u0432\u0433\u0434\u0435\u0451\u0436\u0437\u0438\u0439\u043A\u043B\u043C\u043D\u043E\u043F\u0440\u0441\u0442\u0443\u0444\u0445\u0446\u0447\u0448\u0449\u044C\u044B\u044A\u044D\u044E\u044F",
                "\u0410\u0411\u0412\u0413\u0414\u0415\u0401\u0416\u0417\u0418\u0419\u041A\u041B\u041C\u041D\u041E\u041F\u0420\u0421\u0422\u0423\u0424\u0425\u0426\u0427\u0428\u0429\u042C\u042B\u042A\u042D\u042E\u042F"
            };
````

Reply:
Thanks, will apply later